### ISSUE TITLE
Kerchunk records incorrect calendar type for specific NetCDF files when converted.

### ISSUE THREAD

**Issue by dwest77a** [fsspec/kerchunk#409]
Using example files:

Functions correctly when converted - in "time/.zattrs" calendar is identified as 'noleap':

1. ` https://data.ceda.ac.uk/badc/cmip6/data/CMIP6/CMIP/AS-RCEC/TaiESM1/historical/r1i1p1f1/3hr/clt/gn/v20201013/clt_3hr_TaiESM1_historical_r1i1p1f1_gn_185001010000-185912312230.nc`

Functions incorrectly when converted - calendar identified as '365_day' which causes failure on opening the Kerchunk file with 'decode_times=True', and without this just leaves the time values in float64 format (not good for extracting time stamps):

2. ` https://data.ceda.ac.uk/badc/cmip6/data/CMIP6/CMIP/BCC/BCC-CSM2-MR/historical/r1i1p1f1/Amon/huss/gn/v20181126/huss_Amon_BCC-CSM2-MR_historical_r1i1p1f1_gn_185001-201412.nc`

On creation of the Kerchunk file for the above NetCDF, manually changing the calendar from '365_day' to 'noleap' does solve the issue and the Kerchunk file can be opened with decode_times as True.

Not sure why there is a problem for file 1 but not file 2? I've tried `coo_map={"time": "cf:time"}` but that doesn't seem to make a difference.

**Comment by martindurant** [comment]
I haven't had a chance to look at the data files yet (I'm on slow public wifi) to see what the real difference may be, but it is always possible to postprocess the outputs of the scanning stage (known as preprocessing in the combine stage) to amend any attributes of variables that might be wrong. You will find them stored as JSON at path  "<var>/.zattrs" . Getting that right might require deep knowledge of C-F, but comparison of the two datasets is probably enough.

**Comment by dwest77a** [comment]
Turns out the 'noleap' and '365_day' calendars are pretty much eqivalent and the calendar label was not the issue. The failure in decoding times was an intermittent OverflowError in decoding the times with cftime, probably due to bad data chunks being received (occasionally the float array contains values like 1e-310 instead of 0 which causes the issue). 

I got around this by encoding more of the smaller chunks using a larger inline_threshold, but when writing the chunks in the first instance it could still be an issue of getting bad data, only this time that data is encoded into the Kerchunk file so the decoding will Always fail - but this is just fixed by rerunning the Kerchunk file creation.
